### DISCUSSION TITLE
How Redux uses the Context API

### DISCUSSION THREAD

**Discussion by cpatti97100** [reduxjs/redux#4271] [General]
Hello all and thank you for the great work :)

I read may times that Redux uses the Context API internally. 

I was wondering, is a Redux store handled by a single context?
Or perhaps, does every store slice create its own context?

In other words, what is the magic behind rendering only the components that are affected by a certain part of the state?

Thanks!

**Comment by markerikson** [comment]
To clarify a bit:

_Redux_ doesn't use Context.  The _React-Redux_ library does.

For all the details, see my posts here:

- [The History and Implementation of React-Redux](https://blog.isquaredsoftware.com/2018/11/react-redux-history-implementation/)
- [ReactNext 2019: A Deep Dive into React-Redux](https://blog.isquaredsoftware.com/2019/06/presentation-react-redux-deep-dive/)
- [A (Mostly) Complete Guide to React Rendering Behavior](https://blog.isquaredsoftware.com/2020/05/blogged-answers-a-mostly-complete-guide-to-react-rendering-behavior/)
- [Why React Context is Not a "State Management" Tool (and Why It Doesn't Replace Redux)](https://blog.isquaredsoftware.com/2021/01/context-redux-differences/)
- [React, Redux, and Context Behavior](https://blog.isquaredsoftware.com/2020/01/blogged-answers-react-redux-and-context-behavior/)

**Reply by cpatti97100** [reply]
oh that is correct, sorry for the noise here on the Redux repo Mark :)

I think I can find all I need in the post you linked, fantastic material as usual.

Thank you again!
